reference-lib/src/main/java/org/veriblock/integrations/rewards/PopRewardCalculator.java

```diff
@@ -139,6 +140,29 @@ public static int getBestPublicationHeight(List<AltPublication> publications) th
         return bestPublication;
     }
 
+    public static BigDecimal calculatePopDifficultyForBlock(List<AltChainBlock> blocksInterval) throws SQLException
+    {
+        if(blocksInterval.size() != config.popRewardSettlementInterval + config.popDifficultyAveragingInterval)
+            throw new IllegalArgumentException("The amount of blocks must be equal to popRewardSettlementInterval + popDifficultyAveragingInterval");
+
+        BigDecimal difficulty = BigDecimal.ZERO;
+
+        Collections.sort(blocksInterval);  // make the ascending order for the blocks in the collection, it needs for the correct calculation of the pop score
+
+        for(int i = 0; i < config.popDifficultyAveragingInterval; ++i)
+        {
+            BigDecimal score = calculatePopScoreFromEndorsements(blocksInterval.get(i), blocksInterval.subList(i + 1 , i  + 1 + config.popRewardSettlementInterval));
+            if(score.compareTo(BigDecimal.ZERO) < 1) // less or equal to ZERO
```

Comment:
I think we should compare with BigDecimal.ONE

Reply:
Do we actually need this check? Why cannot we have zero score here?

Reply:
Fixed.